File: src/main/kotlin/cn/solarmoon/spark_core/blueprint/data/BlueprintPin.kt
package cn.solarmoon.spark_core.blueprint.data

data class Pin(
    val id: String,
    val name: String,
    val pinType: PinType,
    val direction: PinDirection? = null,
    val dataType: PinDataType? = null,
    val defaultValue: Any? = null,
    var isConnected: Boolean = false,
    val nodeId: String
)

File: src/main/resources/assets/spark_core/js/blueprint_engine/data/BlueprintPin.js
import { PinType, PinDirection } from './PinEnums.js';
import { PinDataType } from './DataTypes.js';

export class Pin {
    constructor(id, name, pinType, direction = null, dataType = null, defaultValue = null, isConnected = false, nodeId) {
        this.id = id;
        this.name = name;
        this.pinType = pinType;
        this.direction = direction;
        this.dataType = dataType;
        this.defaultValue = defaultValue;
        this.isConnected = isConnected;
        this.nodeId = nodeId;
    }
}
